dpnp/backend/kernels/dpnp_krnl_mathematical.cpp

```diff
@@ -232,6 +232,103 @@ void dpnp_modf_c(void* array1_in, void* result1_out, void* result2_out, size_t s
     event.wait();
 }
 
+template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
+class dpnp_multiply_array_scalar_c_kernel;
+
+template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
+void dpnp_multiply_array_scalar_c(void* input1_in, void* input2_in, void* result_out, const size_t size)
+{
+    if (!size)
+    {
+        return;
+    }
+
+    cl::sycl::event event;
+    _DataType_input1* input1 = reinterpret_cast<_DataType_input1*>(input1_in);
+    _DataType_input2* input2_ptr = reinterpret_cast<_DataType_input2*>(input2_in);
+    _DataType_input2 input2 = *input2_ptr;
+    _DataType_output* result = reinterpret_cast<_DataType_output*>(result_out);
+
+    cl::sycl::range<1> gws(size);
+    auto kernel_parallel_for_func = [=](cl::sycl::id<1> global_id) {
+        size_t i = global_id[0]; /*for (size_t i = 0; i < size; ++i)*/
+        {
+            result[i] = input1[i] * input2;
+        }
+    };
+    auto kernel_func = [&](cl::sycl::handler& cgh) {
+        cgh.parallel_for<class dpnp_multiply_array_scalar_c_kernel<_DataType_input1, _DataType_input2,
+                                                                   _DataType_output>>(gws, kernel_parallel_for_func);
+    };
+    event = DPNP_QUEUE.submit(kernel_func);
+    event.wait();
+}
+
+template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
+class dpnp_multiply_scalar_array_c_kernel;
+
+template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
+void dpnp_multiply_scalar_array_c(void* input1_in, void* input2_in, void* result_out, const size_t size)
+{
+    if (!size)
+    {
+        return;
+    }
+
+    cl::sycl::event event;
+    _DataType_input1* input1_ptr = reinterpret_cast<_DataType_input1*>(input1_in);
+    _DataType_input1 input1 = *input1_ptr;
+    _DataType_input2* input2 = reinterpret_cast<_DataType_input2*>(input2_in);
+    _DataType_output* result = reinterpret_cast<_DataType_output*>(result_out);
+
+    cl::sycl::range<1> gws(size);
+    auto kernel_parallel_for_func = [=](cl::sycl::id<1> global_id) {
+        size_t i = global_id[0]; /*for (size_t i = 0; i < size; ++i)*/
+        {
+            result[i] = input1 * input2[i];
+        }
+    };
+    auto kernel_func = [&](cl::sycl::handler& cgh) {
+        cgh.parallel_for<class dpnp_multiply_scalar_array_c_kernel<_DataType_input1, _DataType_input2,
+                                                                   _DataType_output>>(gws, kernel_parallel_for_func);
+    };
+    event = DPNP_QUEUE.submit(kernel_func);
+    event.wait();
+}
```

Comment:
```suggestion
template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
void dpnp_multiply_scalar_array_c(void* input1_in, void* input2_in, void* result_out, const size_t size)
{
    dpnp_multiply_array_scalar_c<_DataType_input2, _DataType_input1, _DataType_output>(input2_in, input1_in, result_out, size);
}
```
avoid code duplication, please.